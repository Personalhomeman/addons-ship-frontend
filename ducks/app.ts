import { Dispatch } from 'redux';
import { createAction, createReducer } from 'deox';

import { ShipAPIService } from '@/services/ship-api';
import { actionTypeCreator } from '@/utils';
import { App } from '@/models/app';
import { placeholderAppIcon } from '@/config';

const $ = actionTypeCreator('APP');

const _fetchApp = (appSlug: string) => async (
  dispatch: Dispatch,
  _getState: any,
  { shipApi }: { shipApi: ShipAPIService }
) => {
  dispatch(fetchApp.next());

  try {
    const app = await shipApi.getApp(appSlug);

    dispatch(fetchApp.complete(app));
  } catch (error) {
    console.error('fetchApp', error);
    dispatch(fetchApp.error(error));
  }
};

export const fetchApp = Object.assign(_fetchApp, {
  next: createAction($`GET_NEXT`),
  complete: createAction($`GET_COMPLETE`, resolve => (app: App | null) => resolve(app)),
  error: createAction($`GET_ERROR`, resolve => (error: Error) => resolve(error))
});

const defaultState: AppState = null;

export type AppState = App | null;
export default createReducer(defaultState, handleAction => [
  handleAction(fetchApp.complete, (_, { payload }) => {
    if (payload) {
      const avatarUrl = payload.avatarUrl || placeholderAppIcon;
      return { ...payload, avatarUrl } as any;
    }
    return null;
  })
]);
