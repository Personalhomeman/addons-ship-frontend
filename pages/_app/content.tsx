import React, { Fragment, ReactNode } from 'react';
import { connect } from 'react-redux';

import Header from '@/components/Header';
import { RequestError } from '@/models/errors';

import { RootState } from '../../store';
import ErrorPage from '../_error';

type Props = {
  error?: Error | RequestError | null;
  children: ReactNode;
};

export const AppContent = ({ error, children }: Props) => {
  if (error) {
    return (
      <Fragment>
        <Header hideBreadcrumbs />
        <ErrorPage statusCode={(error as RequestError).status || 500} />
      </Fragment>
    );
  }

  return <Fragment>{children}</Fragment>;
};

const mapSateToProps = ({ error }: RootState) => ({
  error
});

// @ts-ignore
export default connect(mapSateToProps)(AppContent);
