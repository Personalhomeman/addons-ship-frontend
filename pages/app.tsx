import { Component, Fragment } from 'react';
import { Dispatch, Store } from 'redux';
import { connect } from 'react-redux';

import { NextContext } from 'next';
import { NextDocumentContext } from 'next/document';

import { AppPageQuery } from '@/models';
import AppSummary from '@/components/AppSummary';
import SectionHeading from '@/components/SectionHeading';

interface AppPageProps extends AppPageQuery {}

type AppPageState = {};

interface Context extends NextDocumentContext {
  store: Store;
  isServer: boolean;
}

class AppPage extends Component<AppPageProps, AppPageState> {
  state: AppPageState = {};

  static getInitialProps({ query: { appSlug } }: NextContext) {
    return { appSlug };
  }

  render() {
    const { appSlug } = this.props;

    return (
      <Fragment>
        <AppSummary
          detailsPagePath="/v1-1-0/details"
          title="My app v1.1.0 (28)"
          description="This is a short description about my app. It is a good app. End of description."
          note="Updated on January 29, 2008"
          iconUrl="/static/icon-flutter.svg"
          platformIconUrl="/static/icon-apple.svg"
        />
        <SectionHeading>Version History</SectionHeading>
      </Fragment>
    );
  }
}

export default AppPage;
