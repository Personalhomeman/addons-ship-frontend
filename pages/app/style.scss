@import '~@/assets/style/mixins';
@import '~@/assets/style/dimensions';
@import '~@/assets/style/colors';

.appPageWrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 7 * $grid-size 7 * $grid-size 18 * $grid-size;

  .appPage {
    width: 100%;
    max-width: 120 * $grid-size;

    .appSummaryWrapper {
      margin-bottom: 6 * $grid-size;
    }

    .majorVersionHeading {
      margin-top: 2 * $grid-size;
      color: $warm-grey;
      // TODO: replace with element from UI library
      line-height: 3 * $grid-size;
      letter-spacing: normal;
      font-size: 19px;
      font-weight: 700;
      font-style: normal;
      font-stretch: normal;
    }
  }

  .footerWrapper {
    margin-top: 15 * $grid-size;
  }
}
