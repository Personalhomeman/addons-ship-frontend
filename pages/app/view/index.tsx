import { Fragment } from 'react';
import formatDate from 'date-fns/format';
import { Flex, Base, Text, Divider, Dropdown } from '@bitrise/bitkit';

import AppSummary from '@/components/AppSummary';
import VersionListPageItem from '@/components/VersionListPageItem';
import Footer from '@/components/Footer';
import { AppVersion } from '@/models';

import css from '../style.scss';

type Props = {
  latestAppVersion: AppVersion | null;
  versionSortingOptions: Array<{
    text: string;
    value: string;
  }>;
  versionSortOptionWithValueSelected: any;
  selectedVersionSortingOption?: {
    text: string;
    value: string;
  };
  groupedAppVersionList: Array<{
    groupName: string;
    appVersions: AppVersion[];
  }>;
};

export default ({
  latestAppVersion,
  versionSortingOptions,
  versionSortOptionWithValueSelected,
  selectedVersionSortingOption,
  groupedAppVersionList
}: Props) => (
  <Flex direction="vertical" className={css.wrapper}>
    <Flex direction="vertical" alignChildrenHorizontal="middle" padding="x16">
      <Base width="100%">
        {latestAppVersion && (
          <AppSummary
            detailsPagePath={`/${latestAppVersion.id}/details`}
            title={`${latestAppVersion.appName} v${latestAppVersion.version} (${latestAppVersion.buildNumber})`}
            description={latestAppVersion.description}
            note={`Updated on ${formatDate(latestAppVersion.lastUpdate, 'MMMM D, YYYY')}`}
            iconUrl={latestAppVersion.iconUrl}
            platformIconUrl="/static/icon-apple.svg"
          />
        )}
        <Base className={css.sectionHeadingWrapper}>
          <Flex direction="horizontal" alignChildrenVertical="end" gap="x2">
            <Flex grow shrink>
              <Text letterSpacing="x1" size="x5" weight="bold">
                Version History
              </Text>
              <Divider color="gray-2" direction="horizontal" margin="x2" />
            </Flex>
            {selectedVersionSortingOption && (
              <Dropdown
                width={168}
                elevation="x1"
                fullWidth={false}
                onChange={versionSortingOptionValue => versionSortOptionWithValueSelected(versionSortingOptionValue)}
                options={versionSortingOptions}
                selected={selectedVersionSortingOption.value}
              >
                {selectedVersionSortingOption.text}
              </Dropdown>
            )}
          </Flex>

          {groupedAppVersionList &&
            groupedAppVersionList.map((appVersionListItem, i) => (
              <Fragment key={i}>
                {selectedVersionSortingOption && selectedVersionSortingOption.value === 'latest-version' && (
                  <Text className={css.majorVersionHeading} size="x4" weight="bold" color="gray-6">
                    v.{appVersionListItem.groupName}
                  </Text>
                )}
                {appVersionListItem.appVersions.map((appVersion, j) => (
                  <VersionListPageItem
                    key={`${i}-${j}`}
                    detailsPagePath={`/apps/${appVersion.appSlug}/versions/${appVersion.id}/details`}
                    platformIconUrl="/static/icon-apple.svg"
                    title={`${appVersion.appName} (${appVersion.buildNumber})`}
                    description={appVersion.description}
                    note={`Updated on ${formatDate(appVersion.lastUpdate, 'MMMM D, YYYY')}`}
                  />
                ))}
              </Fragment>
            ))}
        </Base>
      </Base>
    </Flex>
    <Base paddingVertical="x6">
      <Footer />
    </Base>
  </Flex>
);
