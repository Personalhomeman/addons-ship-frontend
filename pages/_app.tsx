import App, { Container, NextAppContext, DefaultAppIProps } from 'next/app';
import { Store } from 'redux';
import { Provider } from 'react-redux';
import withRedux from 'next-redux-wrapper';
import nookies from 'nookies';
import { ProgressBitbot, Base } from '@bitrise/bitkit';

import { setToken } from '@/ducks/auth';
import Header from '@/components/Header';
import { PageContext } from '@/models';
import { fetchApp } from '@/ducks/app';

import '@/assets/style/index.scss';
import makeStore from '../store';

export interface ShipAppProps extends DefaultAppIProps {
  store: Store;
  appSlug: string;
  token: string;
}

interface AppContext extends NextAppContext {
  ctx: PageContext;
}

export class ShipApp extends App<ShipAppProps> {
  state = {
    ready: false
  };

  static async getInitialProps({ Component, ctx }: AppContext) {
    let { 'auth-token': token } = nookies.get(ctx);
    token = token || 'test-api-token-1';

    const { appSlug } = ctx.query;

    // Set the token on the server side
    if (ctx.isServer) {
      await ctx.store.dispatch(setToken(token) as any);
    }

    if (appSlug) {
      await ctx.store.dispatch(fetchApp(appSlug as string) as any);
    }

    const pageProps = Component.getInitialProps ? await Component.getInitialProps(ctx) : {};

    return { pageProps, appSlug, token };
  }

  async componentDidMount() {
    const { token, store } = this.props;

    // Set token on the client side too
    await store.dispatch(setToken(token) as any);

    this.setState({ ready: true });
  }

  render() {
    const { Component, pageProps, store } = this.props;
    const { ready } = this.state;

    if (!ready) {
      return (
        <Base absolute="center">
          <ProgressBitbot content="Docking Ship..." color="grape-3" />
        </Base>
      );
    }

    return (
      <Container>
        <Provider store={store}>
          <Header />
          <Component {...pageProps} />
        </Provider>
      </Container>
    );
  }
}

export default withRedux(makeStore)(ShipApp);
