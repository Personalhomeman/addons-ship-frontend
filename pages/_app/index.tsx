import App, { Container, NextAppContext, DefaultAppIProps } from 'next/app';
import Router from 'next/router';
import { Store } from 'redux';
import { Provider } from 'react-redux';
import withRedux from 'next-redux-wrapper';
import nookies from 'nookies';
import { ProgressBitbot, Base } from '@bitrise/bitkit';

import { setToken } from '@/ducks/auth';
import Header from '@/components/Header';
import { PageContext } from '@/models';
import { fetchApp } from '@/ducks/app';
import { analyticsConfig } from '@/config';
import { initializeSegment } from '@/utils/analytics';

import '@/assets/style/index.scss';
import makeStore, { RootState } from '../../store';
import AppContent from './content';
import ConfirmEmail from '../confirm';

export interface ShipAppProps extends DefaultAppIProps {
  store: Store;
  appSlug: string;
  token: string;
  settingsOnboardingSeen: boolean;
}

interface AppContext extends NextAppContext {
  ctx: PageContext;
}

export class ShipApp extends App<ShipAppProps> {
  state = {
    ready: false
  };

  static async getInitialProps({ Component, ctx }: AppContext) {
    const cookies = nookies.get(ctx);

    let { appSlug, token } = ctx.query;
    if (!token) {
      token = cookies['api-token'];
    }
    const tokenKey = `token-${appSlug}`;
    if (token && Component.displayName !== ConfirmEmail.displayName) {
      nookies.set(ctx, tokenKey, token as string, {
        maxAge: 1000 * 24 * 60 * 60,
        path: '/'
      });
      nookies.destroy(ctx, 'api-token', {});
    }
    token = token || cookies[tokenKey];

    let { 'settings-onboarding-seen': settingsOnboardingSeen } = cookies;
    settingsOnboardingSeen = settingsOnboardingSeen || 'false';

    // Set the token on the server side
    if (ctx.isServer) {
      await ctx.store.dispatch(setToken(token as string) as any);
    }

    if (appSlug) {
      const {
        app: { app }
      } = ctx.store.getState() as RootState;
      if (!app) {
        await ctx.store.dispatch(fetchApp(appSlug as string) as any);
      }
    } else {
      console.log('No app slug provided');
    }

    const pageProps = Component.getInitialProps ? await Component.getInitialProps(ctx) : {};

    return {
      pageProps,
      appSlug,
      token,
      settingsOnboardingSeen: settingsOnboardingSeen === 'true'
    };
  }

  handleRouteChange = (_url?: string) => {
    const { appSlug, Component } = this.props;
    const pageName = Component.displayName;

    /* istanbul ignore next */
    if (window.analytics) {
      window.analytics.page({ addonId: 'addons-ship', appSlug, pageName });
    }
  };

  async componentDidMount() {
    const { token, store, settingsOnboardingSeen } = this.props;

    if (analyticsConfig.segmentWriteKey) {
      initializeSegment(analyticsConfig.segmentWriteKey);
      this.handleRouteChange();
      Router.events.on('routeChangeComplete', this.handleRouteChange);
    }

    // Set token on the client side too
    await store.dispatch(setToken(token) as any);

    this.setState({ ready: true });

    if (!settingsOnboardingSeen) {
      nookies.set(undefined, 'settings-onboarding-seen', 'true', {
        maxAge: 1000 * 24 * 60 * 60,
        path: '/'
      });
    }
  }

  componentWillUnmount() {
    Router.events.off('routeChangeComplete', this.handleRouteChange);
  }

  render() {
    const { Component, pageProps, store, settingsOnboardingSeen } = this.props;
    const { ready } = this.state;

    if (!ready) {
      return (
        <Base absolute="center">
          <ProgressBitbot content="Docking Ship..." color="grape-3" />
        </Base>
      );
    }

    return (
      <Container>
        <Provider store={store}>
          <AppContent>
            <Header shouldShowSettingsOnboarding={!settingsOnboardingSeen} />
            <Component {...pageProps} />
          </AppContent>
        </Provider>
      </Container>
    );
  }
}

export default withRedux(makeStore)(ShipApp);
