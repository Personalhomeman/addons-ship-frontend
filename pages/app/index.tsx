import { Component } from 'react';
import { connect } from 'react-redux';

import { AppPageQuery, PageContext, AppVersionList } from '@/models';
import { RootState } from '@/store';
import { fetchAppVersionList } from '@/ducks/appVersionList';
import { getAppVersionsByVersion, getAppVersionsByBuildNumber } from '@/ducks/selectors';

import View from './view';

export interface AppPageProps extends AppPageQuery {
  appVersionsByVersion: Array<{
    groupName: string;
    appVersions: AppVersionList;
  }>;
  appVersionsByBuildNumber: Array<{
    groupName: string;
    appVersions: AppVersionList;
  }>;
}

type AppPageState = {
  selectedVersionSortingOptionValue: string | null;
};

export class AppPage extends Component<AppPageProps, AppPageState> {
  state: AppPageState = {
    selectedVersionSortingOptionValue: null
  };

  versionSortingOptions = [
    {
      text: 'Latest Build',
      value: 'latest-build'
    },
    {
      text: 'Latest version',
      value: 'latest-version'
    }
  ];

  componentDidMount() {
    this.setState({
      selectedVersionSortingOptionValue: 'latest-build'
    });
  }

  static async getInitialProps({ query: { appSlug }, store }: PageContext) {
    await store.dispatch(fetchAppVersionList(appSlug as string) as any);

    return { appSlug };
  }

  versionSortOptionWithValueSelected = (value: string) => {
    this.setState({
      selectedVersionSortingOptionValue: value
    });
  };

  render() {
    const groupedAppVersionList =
      this.state.selectedVersionSortingOptionValue && this.state.selectedVersionSortingOptionValue === 'latest-version'
        ? this.props.appVersionsByVersion
        : this.props.appVersionsByBuildNumber;
    const latestAppVersion = groupedAppVersionList[0].appVersions[0];
    const emptyPage = false;

    const viewProps = {
      emptyPage,
      latestAppVersion,
      versionSortingOptions: this.versionSortingOptions,
      versionSortOptionWithValueSelected: this.versionSortOptionWithValueSelected,
      selectedVersionSortingOption: this.versionSortingOptions.find(
        (versionSortingOption: any) => versionSortingOption.value === this.state.selectedVersionSortingOptionValue
      ),
      groupedAppVersionList
    };

    return <View {...viewProps} />;
  }
}

const mapStateToProps = (rootState: RootState) => ({
  appVersionsByVersion: getAppVersionsByVersion(rootState),
  appVersionsByBuildNumber: getAppVersionsByBuildNumber(rootState)
});

export default connect(mapStateToProps)(AppPage as any);
