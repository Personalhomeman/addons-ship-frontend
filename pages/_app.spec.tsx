jest.mock('nookies');
jest.mock('@/ducks/auth');
jest.mock('@/ducks/app');
jest.mock('@/utils/analytics');

import { AppProps } from 'next/app';
import { RouterProps } from 'next/router';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import configureMockStore, { MockStoreCreator, MockStoreEnhanced } from 'redux-mock-store';
import thunk from 'redux-thunk';
import nookies from 'nookies';

import { setToken } from '@/ducks/auth';
import { fetchApp } from '@/ducks/app';
import { initializeSegment } from '@/utils/analytics';

import { ShipApp, ShipAppProps } from './_app';
import { PageContext } from '@/models';

describe('ShipApp', () => {
  let store: MockStoreEnhanced, defaultProps: ShipAppProps & AppProps;
  const setTokenAction = { type: 'MOCK_AUTH_SET_TOKEN' },
    getAppAction = { type: 'MOCK_APP_GET' };

  beforeEach(() => {
    const mockStoreCreator: MockStoreCreator = configureMockStore([thunk]);
    store = mockStoreCreator();

    defaultProps = {
      store,
      appSlug: 'an-app-slug',
      token: 'auth-token',
      settingsOnboardingSeen: false,
      pageProps: {},
      Component: () => <h1>Ship Add-on</h1>,
      router: {} as RouterProps
    };

    ((setToken as any) as jest.Mock).mockReturnValue(setTokenAction);
    ((fetchApp as any) as jest.Mock).mockReturnValue(getAppAction);

    nookies.get = jest.fn().mockImplementation(() => ({}));
  });

  it('renders without errors', () => {
    const tree = shallowToJson(shallow(<ShipApp {...defaultProps} />));

    expect(tree).toMatchSnapshot();
  });

  describe('getInitialProps', () => {
    let spy: jest.SpyInstance;
    const Component = () => <h1>Ship Add-on</h1>;
    Component.getInitialProps = () => ({ page: 'props' });

    let ctx: PageContext;

    beforeEach(() => {
      spy = jest.spyOn(store, 'dispatch');
      ctx = ({
        store,
        isServer: true,
        query: { appSlug: 'an-app-slug', token: 'a-token-from-query' }
      } as any) as PageContext;
    });

    test('on the server', async () => {
      const result = await ShipApp.getInitialProps({ Component, ctx, router: {} as RouterProps });

      expect(result).toMatchSnapshot();
      expect(spy).toHaveBeenCalledWith(setTokenAction);
      expect(spy).toHaveBeenCalledWith(getAppAction);
    });

    test('on the client', async () => {
      const result = await ShipApp.getInitialProps({
        Component,
        ctx: { ...ctx, isServer: false },
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
      expect(spy).not.toHaveBeenCalledWith(setTokenAction);
      expect(spy).toHaveBeenCalledWith(getAppAction);
    });

    test('when settings onboarding has already been seen', async () => {
      nookies.get = jest.fn().mockImplementation(() => ({
        'settings-onboarding-seen': 'true'
      }));

      const result = await ShipApp.getInitialProps({
        Component,
        ctx: { ...ctx, isServer: false },
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
    });

    test('when token is included in query', async () => {
      nookies.set = jest.fn().mockImplementation(() => {});

      const result = await ShipApp.getInitialProps({
        Component,
        ctx: ctx,
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
      expect(nookies.set).toHaveBeenCalledWith(ctx, 'auth-token', 'a-token-from-query', {
        maxAge: 1000 * 24 * 60 * 60,
        path: '/'
      });
    });

    test('when token is not included in query, included in cookie', async () => {
      nookies.get = jest.fn().mockImplementation(() => ({
        'auth-token': 'a-token-from-cookie'
      }));
      nookies.set = jest.fn().mockImplementation(() => {});

      const result = await ShipApp.getInitialProps({
        Component,
        ctx: { ...ctx, query: { appSlug: 'an-app-slug' } },
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
      expect(nookies.set).not.toHaveBeenCalled();
    });

    test('when token is not included in query, nor included in cookie', async () => {
      nookies.set = jest.fn().mockImplementation(() => {});

      const result = await ShipApp.getInitialProps({
        Component,
        ctx: { ...ctx, query: { appSlug: 'an-app-slug' } },
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
      expect(nookies.set).not.toHaveBeenCalled();
    });

    test('when no app slug is present', async () => {
      const result = await ShipApp.getInitialProps({
        Component,
        ctx: { ...ctx, query: {} },
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
      expect(spy).toHaveBeenCalledWith(setTokenAction);
      expect(spy).not.toHaveBeenCalledWith(getAppAction);
    });

    test('when the Component has no getInitialProps method', async () => {
      const result = await ShipApp.getInitialProps({
        Component: () => <h1>Ship Add-on</h1>,
        ctx,
        router: {} as RouterProps
      });

      expect(result).toMatchSnapshot();
    });
  });

  test('componentDidMount', async () => {
    const spy = jest.spyOn(store, 'dispatch');
    const wrapper = await shallow(<ShipApp {...defaultProps} />);

    expect(spy).toHaveBeenCalledWith(setTokenAction);
    expect(store.getActions()).toMatchSnapshot();
    expect(wrapper.state()).toMatchSnapshot();
  });

  describe('analytics', () => {
    it('initializes Segment tracking', () => {
      shallow(<ShipApp {...defaultProps} />);

      expect(initializeSegment).toHaveBeenCalled();
    });

    it('track pageviews on route change', () => {
      const pageEvt = jest.fn(),
        appSlug = 'some-app-slug',
        Component = () => <h1>Hello</h1>;
      Component.displayName = 'Whatever';
      Object.defineProperty(window, 'analytics', { value: { page: pageEvt } });

      const wrapper = shallow(<ShipApp {...defaultProps} appSlug={appSlug} Component={Component as any} />);

      (wrapper.instance() as ShipApp).handleRouteChange('some-url');
      expect(pageEvt).toHaveBeenCalledWith({ addonId: 'addons-ship', appSlug, pageName: Component.displayName });
    });
  });
});
