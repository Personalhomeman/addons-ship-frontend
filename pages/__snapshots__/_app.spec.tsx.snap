// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipApp componentDidMount 1`] = `
Array [
  Object {
    "type": "MOCK_AUTH_SET_TOKEN",
  },
]
`;

exports[`ShipApp componentDidMount 2`] = `
Object {
  "ready": true,
}
`;

exports[`ShipApp getInitialProps on the client 1`] = `
Object {
  "appSlug": "an-app-slug",
  "pageProps": Object {
    "page": "props",
  },
  "token": "test-api-token-1",
}
`;

exports[`ShipApp getInitialProps on the server 1`] = `
Object {
  "appSlug": "an-app-slug",
  "pageProps": Object {
    "page": "props",
  },
  "token": "test-api-token-1",
}
`;

exports[`ShipApp getInitialProps when no app slug is present 1`] = `
Object {
  "appSlug": undefined,
  "pageProps": Object {
    "page": "props",
  },
  "token": "test-api-token-1",
}
`;

exports[`ShipApp getInitialProps when the Component has no getInitialProps method 1`] = `
Object {
  "appSlug": "an-app-slug",
  "pageProps": Object {},
  "token": "test-api-token-1",
}
`;

exports[`ShipApp renders without errors 1`] = `
<Base
  absolute="center"
>
  <ProgressBitbot
    color="grape-3"
    content="Docking Ship..."
    size="64px"
  />
</Base>
`;
